### PR TITLE
Super-draft version of experiment-runner concept proposal

### PR DESCRIPTION
**Pull request by hexfaker**
This is a WIP proposal of new catalyst model. The main idea is to separate objects that describe how to run experiment (Experiments) and objects that perform them (Runners).

Right now it's in very early state, so not look at it!!!

### CODE REVIEW

File: examples/experimental/simple_cifar.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        x = self.pool(F.relu(self.conv1(x)))
+        x = self.pool(F.relu(self.conv2(x)))
+        x = x.view(-1, 16 * 5 * 5)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+
+class VerboseCallback(Callback):
```
Comment (TezRomacH) [inline examples/experimental/simple_cifar.py:92]: looks nice! Let's move to `catalyst.callbacks.logger`

File: catalyst/dl/experiments/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from .experiment import Experiment
```
Comment (TezRomacH) [inline catalyst/dl/experiments/__init__.py:1]:
```suggestion
# flake8: noqa
from .experiment import *
```

File: catalyst/dl/experiments/experiment.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        pass
+
+    def get_state_params(self, stage: str) -> Mapping[str, Any]:
+        return dict(
+            logdir=self.logdir,
+            total_epochs=self.get_total_epochs(stage)
+        )
+
+
+class SimpleExperiment(Experiment):
```
Comment (TezRomacH) [inline catalyst/dl/experiments/experiment.py:69]: -> `CommonExperiment` ?

File: catalyst/dl/experiments/experiment.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    @abstractmethod
+    def get_optimizer(self, stage: str) -> _Optimizer:
+        pass
+
+    @abstractmethod
+    def get_total_epochs(self, stage: str):
+        pass
+
+    @abstractmethod
+    def get_loaders(self, stage: str) -> "OrderedDict[str, DataLoader]":
```
Comment (TezRomacH) [inline catalyst/dl/experiments/experiment.py:47]: > "OrderedDict[str, DataLoader]"

why in quotes?

**Review by TezRomacH — APPROVED** [review]

**Comment by Scitator** [comment]
moved to 19.02 proposal
https://github.com/catalyst-team/catalyst/pull/103

**Pull request closed without merging**